cmd/kubeadm/app/discovery/file/file.go

```diff
@@ -140,6 +102,54 @@ func ValidateConfigInfo(config *clientcmdapi.Config, clustername string) (*clien
 	return refreshedBaseKubeConfig, nil
 }
 
+// get kubeconfig from config
+func getKubeconfig(config *clientcmdapi.Config, defaultCluster *clientcmdapi.Cluster, clustername string) (*clientcmdapi.Config, error) {
+	// Create a new kubeconfig object from the given, just copy over the server and the CA cert
+	// We do this in order to not pick up other possible misconfigurations in the clusterinfo file
+	kubeconfig := kubeconfigutil.CreateBasic(
+		defaultCluster.Server,
+		clustername,
+		"", // no user provided
+		defaultCluster.CertificateAuthorityData,
+	)
+
+	// load pre-existing client certificates
+	if config.Contexts[config.CurrentContext] != nil && len(config.AuthInfos) > 0 {
+		user := config.Contexts[config.CurrentContext].AuthInfo
+		authInfo, ok := config.AuthInfos[user]
+		if !ok || authInfo == nil {
+			return nil, errors.Errorf("empty settings for user %q", user)
+		}
+		if len(authInfo.ClientCertificate) != 0 {
```

Comment:
validateKubeConfig has ensured that len(authInfo.ClientKeyData)  must be 0 when len(authInfo.ClientCertificate) != 0, so here needn't to judge whether len(authInfo.ClientKeyData) is zero or not.

Reply:
I might be missing something but based on `clientcmd.Validate` what I see is that it excludes `ClientKey` and `ClientKeyData` from both being provided (or none), as well as `ClientCertificate` and `ClientCertificateData`, but I don't see this relationship that `len(authInfo.ClientCertificate) != 0 => len(authinfo.ClientKeyData) == 0`.

Reply:
Sorry, there are typos in my comment, it should be following:
validateKubeConfig has ensured that len(authInfo.ClientCertificateData) must be 0 when len(authInfo.ClientCertificate) != 0, so here needn't to judge whether len(authInfo.ClientCertificateData) is zero or not.

if len(authInfo.ClientCertificateData) == 0 && len(authInfo.ClientCertificate) != 0 

I delete "len(authInfo.ClientCertificateData) == 0".